webapp/src/components/custom_post_types/post_type_recording.tsx

```diff
@@ -1,8 +1,73 @@
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
+import {useSelector} from 'react-redux';
+import styled from 'styled-components';
+
+import {GlobalState} from '@mattermost/types/store';
+import IconAI from 'src/components/icons/ai';
+
+const aiPluginID = 'mattermost-ai';
+
+const useAIAvailable = () => {
+    //@ts-ignore plugins state is a thing
+    return useSelector<GlobalState, boolean>((state) => Boolean(state.plugins?.plugins?.[aiPluginID]));
+};
```

Comment:
Considering this is a straight lookup, we could add it to the selectors and cut out two functions (which would have to be rerendered on every global state change since react can't tell from the `useSelector` what part of global state we're interested in).

Reply:
Not sure what you are suggesting. I think that useSelector will take care of not triggering a re-render if the Boolean result or the function have not changed. Am I reading this wrong? https://react-redux.js.org/api/hooks#useselector

Reply:
You're right!